### ISSUE TITLE
Flutter Web Content Type behavior acting differently than other platforms

### ISSUE THREAD

**Issue by izio38** [flutterchina/dio#994]
### New Issue Checklist

- [x] I have searched for a similar issue in the [project](https://github.com/flutterchina/dio/issues) and found none

### Issue Info

| Info                           | Value                                                 |      |
| ------------------------------ | ----------------------------------------------------- | ---- |
| Platform Name                  | Flutter Web                          |      |
| Platform Version               |    Flutter 1.23.0-18.1.pre • channel beta                           |      |
| Dio Version                    | 3.0.10 (latest)                                |      |
| Android Studio / Xcode Version | Visual Studio Insider              |      |
| Repro rate                     | all the time   |      |
| Repro with our demo prj        | I have not checked            |      |
| Demo project link              | Not needed |      |

### Issue Description and Steps

We use Dio client in our Flutter app, which build both android and web version. On the web version there is a different behavior than other platforms.

In fact, if we want a post request to have a json payload (body), we pass a `Content-Type` = `application/json`. With this header, body is well filled in other platforms, but in Web, the body stays empty, whatever json we pass to the post request.

If we change to `content-type` (without upper case), it works for all platforms.

To summarize : 
- Setting this header : `Content-Type` = `application/json` makes the body empty for post request whatever we pass.

Thanks.

**Comment by izio38** [comment]
Why this bot exist ?
